### PR TITLE
Factor out etcdWatcher and rename methods in StorageInterface

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #10475 

Follow-up from #11666

### CODE REVIEW

File: pkg/tools/etcd_helper_test.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func TestExtractObj(t *testing.T) {
 	}
 }
 
-func TestExtractObjNotFoundErr(t *testing.T) {
+func TestGetjNotFoundErr(t *testing.T) {
```
Comment (derekwaynecarr) [inline pkg/tools/etcd_helper_test.go]: TestGetNotFoundErr

**Comment by derekwaynecarr** [comment]
Nit then lgtm

File: pkg/tools/etcd_helper_test.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/tools/etcd_helper_test.go]: Done.

**Comment by wojtek-t** [comment]
@derekwaynecarr - since you already reviewed it - can you please apply LGTM label?

**Comment by derekwaynecarr** [comment]
@wojtek-t - done

**Comment by wojtek-t** [comment]
@derekwaynecarr - great - thanks!

**Pull request merged**